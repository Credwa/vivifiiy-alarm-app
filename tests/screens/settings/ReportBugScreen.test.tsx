import ReportBugScreen from '@/screens/settings/ReportBugScreen';
import { AntDesign } from '@expo/vector-icons';
import * as Font from 'expo-font';
import React from 'react';
import renderer from 'react-test-renderer';
jest.useFakeTimers();

describe('<ReportBugScreen />', () => {
  beforeEach(async () => {
    await Font.loadAsync({
      ...AntDesign.font
    });
  });

  it(`renders correctly`, async () => {
    const mockNav: any = { goBack: jest.fn() };
    const mockRoute: any = { params: {} };
    let tree = await renderer.create(<ReportBugScreen navigation={mockNav} route={mockRoute} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
