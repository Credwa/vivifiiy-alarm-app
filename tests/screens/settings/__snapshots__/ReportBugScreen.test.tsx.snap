// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportBugScreen /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RNCSafeAreaView
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 25,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignContent": "space-around",
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "marginBottom": 50,
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "flex-start",
              "alignItems": "flex-start",
              "height": 44,
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CEE1F2",
                  "fontSize": 28,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#869DB2",
              },
              Object {
                "fontFamily": "Nunito_400Regular",
                "fontSize": 20,
              },
              Object {
                "color": "#CEE1F2",
              },
              Object {
                "marginLeft": -15,
              },
            ]
          }
        >
          Report bug
        </Text>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CEE1F2",
                  "fontSize": 28,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
